src/net_processing.cpp

```diff
@@ -3192,7 +3402,13 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
             // requests for it.
             m_txrequest.ForgetTxHash(tx.GetHash());
             m_txrequest.ForgetTxHash(tx.GetWitnessHash());
-            RelayTransaction(tx.GetHash(), tx.GetWitnessHash(), m_connman);
+
+            // Flood those transactions which were received either via flooding, or inbound reconciliation,
+            // but NOT via outbound reconciliation. Flooding then is mainly used for initial propagation
+            // of new transactions across a network of reachable nodes quickly.
+            LOCK(peer->m_recon_state_mutex);
+            bool flood = !(peer->m_recon_state && peer->m_recon_state->m_responder);
```

Comment:
Use a `WITH_LOCK()` macro or code block here to avoid holding the mutex for the rest of this block.